src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/BlobManager.java

```diff
@@ -84,10 +85,22 @@ public CloudBlobContainer getContainer(String containerName) throws URISyntaxExc
     }
 
     public List<CloudBlobContainer> listInputContainers() {
-        return StreamSupport
+        List<CloudBlobContainer> cloudBlobContainerList = StreamSupport
             .stream(cloudBlobClient.listContainers().spliterator(), false)
             .filter(c -> !c.getName().endsWith(REJECTED_CONTAINER_NAME_SUFFIX))
+            .filter(this::filterBySelectedContainer)
             .collect(toList());
+
+        if (cloudBlobContainerList.isEmpty()) {
+            log.error("Container not found for configured container name : {}", properties.getBlobSelectedContainer());
+        }
+
+        return cloudBlobContainerList;
+    }
+
+    private boolean filterBySelectedContainer(CloudBlobContainer container) {
+        return SELECT_ALL_CONTAINER.equalsIgnoreCase(properties.getBlobSelectedContainer())
+            || properties.getBlobSelectedContainer().equals(container.getName());
```

Comment:
`properties.getBlobSelectedContainer()` is repeated thrice could we please extract it ?

Reply:
> `properties.getBlobSelectedContainer()` is repeated thrice could we please extract it ?

changed